File: implementations/java/compression/src/pipelines/RunLogging.java
package implementations.java.compression.src.pipelines;

/** Gate for verbose stdout during a pipeline run; set from {@link PipelineParams#debug()}. */
public final class RunLogging {

    private static boolean debug;

    private RunLogging() {
    }

    public static void setDebug(boolean debug) {
        RunLogging.debug = debug;
    }

    public static boolean isDebug() {
        return debug;
    }

    public static void debug(String message) {
        if (debug) {
            System.out.println(message);
        }
    }
}

File: implementations/cpp/utils/run_logging.cpp
#include "run_logging.hpp"

#include <iostream>

namespace run_logging {

namespace {
bool g_debug = false;
}

void set_debug(bool debug) {
    g_debug = debug;
}

bool is_debug() {
    return g_debug;
}

void debug(const std::string& message) {
    if (g_debug) {
        std::cout << message << '\n';
    }
}

}  // namespace run_logging
